### ISSUE TITLE
Join POI to the contour of the building makes its icon as the point

### ISSUE THREAD

**Issue by Andygol** [openstreetmap/iD#3425]
Does it possible to leave POI icon as it is?

![poi](https://cloud.githubusercontent.com/assets/369696/18566420/85b48b62-7b9c-11e6-826b-c1ead9a9abb9.gif)

/cc @bhousel

**Comment by Andygol** [comment]
☝️ based on the claim on the forum http://forum.openstreetmap.org/viewtopic.php?pid=608812#p608812

**Comment by bhousel** [comment]
iD's presets each have an expected geometry that they are supposed to match.  For POI presets like shops, the allowable geometry includes "point" (standalone node) and "area" (closed way), but not "vertex" (node along a way).  So that's why moving the shop onto the line means that it's no longer recognized as the shop preset nor shown with a shop icon.

Not sure if there is much that we should do here.  If we allow all the amenities, shops, etc to have vertex geometry, it means that these features will be suggested in a lot of places that they really don't belong - like along highways, railways, and so on.

**Comment by jleedev** [comment]
It seems to me that the desired behavior is to move the tags from the node to the area, and discard the node, or else to move the node to the interior of the area, but never to create a vertex.

Quite often I've tried to draw building outlines using iD and I have found myself manually entering all of the existing tags from the node and then deleting it. I suspect that JOSM has some smarts here.

Another approach would be some command to expand a node into a small triangle or rectangle for ease of editing; again, disallow creating a vertex since that's not a proper approach.

**Comment by Andygol** [comment]
TL;DR (translation from Russian) - https://github.com/openstreetmap/iD/issues/3425#issuecomment-247450871
Main disadvantage is in the fact that vertex don't allow to see icon of the POI. And then people just double it on a separate point inside of the building polygon. Does it possible to show it on closer zoom level?

**Comment by bhousel** [comment]
> And then people just double it on a separate point inside of the building polygon

A lot of people think that businesses and shops _should_ be mapped as separate points inside building polygons  (and that points along building polygons should be `entrance=*` nodes).

Regardless of what people think.. We could split up `vertex` geometry type into 2 separate geometry types:  `areaVertex` and `lineVertex`, this way some features (like entrances) could appear on the perimeters of areas, and other features (like traffic signals) could appear only on linear ways.

**Comment by d1g** [comment]
> desired behavior is to move the tags from the node to the area

@jleedev, we want to keep history first. There is no redirects in OSM, it is possible to _damage_ history of edits. If we change API to support redirects then we could support only one way of doing it.

Wikidata prefers older ids during merges, same should be done in OSM (see osmwiki).

> points along building polygons should be entrance=* nodes

@bhousel, sometimes there is no door for customers and some mappers (incl. myself) place shop= node where window is located.

kiosk is a typical example of building without door for customers.

**Comment by bhousel** [comment]
For now I'm going to continue to encourage people to map this way:
- buildings/structures as areas 
- businesses/tenants as points
- entrances/addresses as vertices along the building/structure edge.

This is easier than trying to guess whether a POI can be used as vertex based on the tags of the dragged thing and the tags of the thing it got dragged onto.
